tests/robots/test_robot.py

```diff
@@ -13,54 +18,160 @@
+    return Configuration.from_revolute_values([math.pi/2, 0., 0., math.pi/4, 0., 0.])
+
+
+@pytest.fixture
+def panda_configuration():
+    return Configuration.from_revolute_values([math.pi/4, math.pi/3., 0., math.pi/4, 0., 0., math.pi/2, 0.])
 
 
 def test_basic_name_only():
-    robot = Robot.basic('testbot')
+    robot = Robot.basic("testbot")
     assert robot.artist is None
 
 
 def test_basic_name_joints_links(ur5_joints, ur5_links):
-    robot = Robot.basic('testbot', joints=ur5_joints, links=ur5_links)
+    robot = Robot.basic("testbot", joints=ur5_joints, links=ur5_links)
     assert len(robot.model.links) == 11
     assert len(robot.get_configurable_joint_names()) == 6
 
 
 def test_basic_name_joints(ur5_joints):
-    robot = Robot.basic('testbot', joints=ur5_joints)
+    robot = Robot.basic("testbot", joints=ur5_joints)
     assert len(robot.model.joints) == 10
 
 
 def test_basic_name_links(ur5_links):
-    robot = Robot.basic('testbot', links=ur5_links)
+    robot = Robot.basic("testbot", links=ur5_links)
     assert len(robot.model.links) == 11
 
 
 def test_basic_attr():
-    robot = Robot.basic('testbot', location="rfl")
-    assert robot.model.attr['location'] == "rfl"
+    robot = Robot.basic("testbot", location="rfl")
+    assert robot.model.attr["location"] == "rfl"
+
+
+def test_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.name == "panda"
+
+
+def test_group_names(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.group_names == ["manipulator", "endeffector"]
+
+
+def test_main_group_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.main_group_name == "panda_arm_hand"
+
+
+def test_root_name(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.root_name == "world"
+
+
+@pytest.mark.parametrize("group,expected_result", [(None, "panda_rightfinger"), ("panda_arm_hand", "panda_rightfinger"), ("panda_arm", "panda_link8")])
+def test_get_end_effector_link_name(panda_robot_instance, group,  expected_result):
+    robot = panda_robot_instance
+    assert robot.get_end_effector_link_name(group=group) == expected_result
+
+
+def test_get_end_effector_link_name_wrong_group(panda_robot_instance):
+    robot = panda_robot_instance
+    with pytest.raises(KeyError):
+        robot.get_end_effector_link_name(group="panda_leg")
+
+
+@pytest.mark.parametrize("group", [None, "endeffector"])
+def test_get_end_effector_link(ur5_robot_instance, group):
+    robot = ur5_robot_instance
+    assert isinstance(robot.get_end_effector_link(group=group), compas.robots.Link)
+
+
+@pytest.mark.parametrize("group, x_coord", [(None, .301), ("panda_arm", .359)])
+def test_get_end_effector_frame(panda_robot_instance, group, x_coord):
+    robot = panda_robot_instance
+    ee_frame = robot.get_end_effector_frame(group=group)
+    assert round(ee_frame.point.x, 3) == x_coord
+
+
+@pytest.mark.parametrize("group, remove_semantics, expected_result",
```

Comment:
where is `remove_semantics` coming from here?

Reply:
The list given as the last argument contains tuples of three. For every run (one per tuple) `group`, `remove_semantics` and `expected_result` is set to the values in the tuple and used as arguments.

Reply:
If the question is why I remove semantics in the tests I can't say I remember for sure.

Reply:
ah sorry, I got this one now ;)